components/Tabs/Tabs.js

```diff
@@ -0,0 +1,83 @@
+
+class TabsItem {
+  constructor(element) {
+    // attach dom element to object. Example in Tabs class
+    this.element = element;
+
+  }
+
+  select() {
+    // should use classList
+    this.element.classList.add("Tabs__item-selected");
+  }
+
+  deselect() {
+    // should use classList
+    this.element.classList.remove("Tabs__item-selected");
+  }
+}
+
+class TabsLink {
+  constructor(element, parent) {
+    this.element = element;// attach dom element to object
+    this.tabs = parent;// attach parent to object
+    this.tabsItem = parent.getTab(element.dataset.tab);// assign this to the associated tab using the parent's "getTab" method by passing it the correct data
+    this.tabsItem = new TabsItem(this.tabsItem);
+    // reassign this.tabsItem to be a new instance of TabsItem, passing it this.tabsItem
+    this.element.addEventListener('click', () => {
+      this.tabs.updateActive(this);
+      this.select();
+    });
+  };
+
+  select() {
+    // select this link
+    // select the associated tab
+    this.element.classList.add("Tabs__link-selected");
+    this.tabsItem.select();
+  }
+
+  deselect() {
+    // deselect this link
+    // deselect the associated tab
+    this.element.classList.remove("Tabs__link-selected");
+    this.tabsItem.deselect();
+  }
+}
+
+class Tabs {
+  constructor(element) {
+    this.element = element;// attaches the dom node to the object as "this.element"
+    this.links = element.querySelectorAll(".Tabs__link");
+    this.links = Array.from(this.links).map((link) => {
+      return new TabsLink(link, this);
+    });
+    this.activeLink = this.links[0];
+    this.init();
+  }
+
+  init() {
+    // select the first link and tab upon ititialization
+    this.activeLink.select();
+  }
+
+  updateActive(newActive) {
+    // deselect the old active link
+    // assign the new active link
+    this.activeLink.deselect();
+    this.activeLink = newActive
+  }
+
+  getTab(data) {
+    // use the tab item classname and the data attribute to select the proper tab
+    let allTabs = document.querySelectorAll(".Tabs__item");
```

Comment:
This is awesome. You could also do this in one line with this:

`return this.element.querySelector(`.Tabs__item[data-tab="${data}"]`);`

Reply:
This is what I originally did, but it kept giving me errors so I went for the quick fix